### PR TITLE
Fixup file type names.

### LINKED ISSUE #93999: Mono WASM PInvoke Table generation does not handle file scoped types
### Description

File scoped types are named like `<File_Name_Without_Periods>Hash__ClassName`. For `UnmanagedCallersOnly` methods in file-scoped types, Mono tries to generate a method in C, but it doesn't escape the name. As a result, you get errors like the following:

```
/root/helix/work/correlation/build/wasm/WasmApp.Native.targets(367,5): error : In file included from /root/helix/work/correlation/build/microsoft.netcore.app.runtime.browser-wasm/runtimes/browser-wasm/native/src/pinvoke.c:15: [/root/helix/work/workitem/e/publish/ProxyProjectForAOTOnHelix.proj]
/root/helix/work/correlation/build/wasm/WasmApp.Native.targets(367,5): error : /root/helix/work/workitem/e/wasm_build/obj/wasm/for-build/pinvoke-table.h:212:64: error: expected ';' after top level declarator [/root/helix/work/workitem/e/publish/ProxyProjectForAOTOnHelix.proj]
/root/helix/work/correlation/build/wasm/WasmApp.Native.targets(367,5): error : int wasm_native_to_interp_System_Runtime_InteropServices_Tests_<System_Runtime_InteropServices_Tests_ComVariantMarshallerTests_IGeneratedComInterface>F3F2786D8F8679299059CA9E93026F23079CBD4B40DBBE7EBFBCA0FE1F6116567__InterfaceImplementation_ABI_Method (int arg0) {  [/root/helix/work/workitem/e/publish/ProxyProjectForAOTOnHelix.proj]
/root/helix/work/correlation/build/wasm/WasmApp.Native.targets(367,5): error :                                                                ^ [/root/helix/work/workitem/e/publish/ProxyProjectForAOTOnHelix.proj]
/root/helix/work/correlation/build/wasm/WasmApp.Native.targets(367,5): error :                                                                ; [/root/helix/work/workitem/e/publish/ProxyProjectForAOTOnHelix.proj]
/root/helix/work/correlation/build/wasm/WasmApp.Native.targets(367,5): error : /root/helix/work/workitem/e/wasm_build/obj/wasm/for-build/pinvoke-table.h:220:3: error: use of undeclared identifier 'WasmInterpEntrySig_5' [/root/helix/work/workitem/e/publish/ProxyProjectForAOTOnHelix.proj]
/root/helix/work/correlation/build/wasm/WasmApp.Native.targets(367,5): error : ((WasmInterpEntrySig_5)wasm_native_to_interp_ftndescs [5].func) ((int*)&res, (int*)&arg0, (int*)&arg1, wasm_native_to_interp_ftndescs [5].arg); [/root/helix/work/workitem/e/publish/ProxyProjectForAOTOnHelix.proj]
/root/helix/work/correlation/build/wasm/WasmApp.Native.targets(367,5): error :   ^ [/root/helix/work/workitem/e/publish/ProxyProjectForAOTOnHelix.proj]
/root/helix/work/correlation/build/wasm/WasmApp.Native.targets(367,5): error : /root/helix/work/workitem/e/wasm_build/obj/wasm/for-build/pinvoke-table.h:269:420: error: use of undeclared identifier 'System_Runtime_InteropServices_Tests_ComVariantMarshallerTests_IGeneratedComInterface' 
```

### Reproduction Steps

Build https://github.com/dotnet/runtime/pull/93635 in the LibraryTests_EAT configuration.

### Expected behavior

Build succeeds

### Actual behavior

Build fails with above mentioned error.

### Regression?

Unknown

### Known Workarounds

_No response_

### Configuration

Running in a PR into this repo: https://github.com/dotnet/runtime/pull/93635

### Other information

Helix log for the failing test:

https://[helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-93635-merge-9667f32037a34a4fa0/normal-System.Runtime.InteropServices.Tests/1/console.147620a6.log?helixlogtype=result](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-93635-merge-9667f32037a34a4fa0/normal-System.Runtime.InteropServices.Tests/1/console.147620a6.log?helixlogtype=result)

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Fixes #93999

### CODE REVIEW

**Review by AaronRobinsonMSFT — APPROVED** [review]

File: src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs [hunk 1]
```diff
@@ -308,7 +308,7 @@ private void EmitNativeToInterp(StreamWriter w, List<PInvokeCallback> callbacks)
 
             string module_symbol = _fixupSymbolName(method.DeclaringType!.Module!.Assembly!.GetName()!.Name!);
             uint token = (uint)method.MetadataToken;
-            string class_name = method.DeclaringType.Name;
+            string class_name = _fixupSymbolName(method.DeclaringType.Name);
             string method_name = method.Name;
             string entry_name = $"wasm_native_to_interp_{module_symbol}_{class_name}_{method_name}";
```
Comment (radical) [inline src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs:313]: Maybe the fixup should be done on `entry_name`, so we cover the whole symbol name.

File: src/mono/wasm/Wasm.Build.Tests/PInvokeTableGeneratorTests.cs [hunk 2]
```diff
@@ -375,6 +375,39 @@ public static int Main()
...
+                }
+
+                file class Foo
+                {
+                    [UnmanagedCallersOnly]
+                    public unsafe static extern void SomeFunction1(int i);
+                }
+                """;
+
+            (_, string output) = BuildForVariadicFunctionTests(
```
Comment (radical) [inline src/mono/wasm/Wasm.Build.Tests/PInvokeTableGeneratorTests.cs:402]: We should run this also to ensure that any references to such symbols are correct, and usable.

File: src/mono/wasm/Wasm.Build.Tests/PInvokeTableGeneratorTests.cs [hunk 3]
```diff
@@ -375,6 +375,42 @@ public static int Main()
             Assert.Matches("warning\\sWASM0001.*Skipping.*Test::SomeFunction1.*because.*function\\spointer", output);
         }
 
+        [Theory]
+        [BuildAndRun(host: RunHost.Chrome)]
+        public void UnmanagedCallback_InFileType(BuildArgs buildArgs, string id)
```
Comment (jkoritzinsky) [inline src/mono/wasm/Wasm.Build.Tests/PInvokeTableGeneratorTests.cs:380]:
```suggestion
        public void UnmanagedCallback_InFileType(BuildArgs buildArgs, RunHost host, string id)
```

**Review by radical — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
Failure is unrelated. Merging this in.

**Pull request merged**